edweiss-app/__test__/components/courses/SelectActionsCourse.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { fireEvent, render } from '@testing-library/react-native';
+import { TextProps, TouchableOpacityProps, ViewProps } from 'react-native';
+
+
+jest.mock('@/components/core/containers/TView.tsx', () => {
+    const { View } = require('react-native');
+    return (props: ViewProps) => <View {...props} />;
+});
+
+jest.mock('@/components/core/TText.tsx', () => {
+    const { Text } = require('react-native');
+    return (props: TextProps) => <Text {...props} />;
+});
+
+jest.mock('@/components/core/containers/TTouchableOpacity.tsx', () => {
+    const { TouchableOpacity, View } = require('react-native');
+    return (props: React.PropsWithChildren<TouchableOpacityProps>) => (
+        <TouchableOpacity {...props}>
+            <View>{props.children}</View>
+        </TouchableOpacity>
+    );
+});
+
+jest.mock('@/components/core/Icon', () => {
+    return {
+        __esModule: true,
+        default: ({ name, size = 16, color = 'subtext0', testID }: { name: string; size?: number; color?: string; testID?: string }) => {
+            const { Text } = require('react-native');
+            return <Text testID={testID || 'icon'}>{`Icon - ${name} - Size: ${size} - Color: ${color}`}</Text>;
+        }
+    };
+});
+
+// Mock t() function
+jest.mock('@/config/i18config', () =>
+    jest.fn((str: string) => {
+        if (str === 'course:add_assignment') return 'Add Assignment';
+        else if (str === 'course:add_material') return 'Add Material';
+        else return str;
+    })
+);
+
+
+describe('SelectActionsCourse', () => {
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const mockOnOutsideClick = jest.fn();
+    const mockOnSelectAssignment = jest.fn();
+    const mockOnSelectMaterial = jest.fn();
+
+    it('should render the component', () => {
+        const screen = render(<SelectActionsCourse
+            onOutsideClick={mockOnOutsideClick}
+            onSelectAssignment={mockOnSelectAssignment}
+            onSelectMaterial={mockOnSelectMaterial}
+        />);
+
+        expect(screen.getByText('Add Assignment')).toBeTruthy();
+        expect(screen.getByText('Add Material')).toBeTruthy();
+
+        expect(screen.getByTestId(testIDs.screenTouchable)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.screenView)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.addAssignmentBouton)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.addAssignmentText)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.addMaterialBouton)).toBeTruthy();
+        expect(screen.getByTestId(testIDs.addMaterialText)).toBeTruthy();
+    });
+
+    it('should call onOutsideClick when the screenTouchable is pressed', () => {
+        const screen = render(<SelectActionsCourse
+            onOutsideClick={mockOnOutsideClick}
+            onSelectAssignment={mockOnSelectAssignment}
+            onSelectMaterial={mockOnSelectMaterial}
+        />);
```

Comment:
I see that you reuse a lot of this boilerplate code. 
I propose a create a renderScreen where you put the code and then call it inside.
 Here's my suggestion:

``` tsx
const renderScreen =  render(<SelectActionsCourse
            onOutsideClick={mockOnOutsideClick}
            onSelectAssignment={mockOnSelectAssignment}
            onSelectMaterial={mockOnSelectMaterial}
        />);
```

and then you only have to call it in the tests. This is a way of improvement, feel free to do it, if you feel like it make sens to you in your code base.